### DISCUSSION TITLE
Logs when running as service?

### DISCUSSION THREAD

**Discussion by vbfischer** [FelixKratz/SketchyBar#335] [Q&A] [answered]
I have a plugin I am writing and it works great when I run sketchybar from command line:

```
sketchybar
```

it shows up:
![CleanShot 2023-03-03 at 09 37 36](https://user-images.githubusercontent.com/138106/222748265-98964046-6c0e-4c71-8425-a2677e15302d.png)

But its not displaying the content when I run as a brew service:

```
brew services start sketchybar
```
![CleanShot 2023-03-03 at 09 38 16](https://user-images.githubusercontent.com/138106/222748384-62b8013d-b9c5-42a5-9c35-3ec6950c7ef6.png)

**Comment by vbfischer** [comment, marked as answer]
I think I figured it out... I'm guessing the service doesn't have access to environment variables...  

With that in mind, any suggestions on how to store stuff like API Keys when you plan to store your config in git?

**Comment by FelixKratz** [comment]
Something very simple would be to store it in a file (e.g. define it as a variable in a .sh file) and add this file to .gitignore, then source the file and the variable will be accessible in your scripts. the sketchybar service is a clean environment with only the most important env variables set e.g. `$USER` and `$HOME` etc.

Maybe I will add something to sketchybar when run from the commandline to drop the current session environment variables per default.

**Comment by OJFord** [comment]
`/opt/homebrew/var/log/sketchybar/sketchybar.err.log`

(I needed them for a different problem, 'it's a different environment, put it in a file' is an 'XY answer' as it were, didn't help me.)

**Reply by robrecord** [reply]
On intel machines, it's `/usr/local/var/log/sketchybar/sketchybar.err.log` - also there's a stdout log in that directory too.
